### PR TITLE
Support managed identity in BRS for control 'Azure_AppService_DP_Dont_Allow_HTTP_Access'

### PR DESCRIPTION
**Pull request by Aditi-sha**

### CODE REVIEW

File: Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1 [hunk 1]
```diff
@@ -9,7 +9,8 @@
     Use HTTPS for App Services.
 
 # Prerequisites:
-    Contributor and higher privileges on the App Services in a Subscription.
+    1. Contributor and higher privileges on the App Services in a Subscription.
+    2. AzAccount must be connected.
```
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1:13]: Must be connected to Azure with an authenticated account

File: Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1 [hunk 2]
```diff
@@ -9,7 +9,8 @@
     Use HTTPS for App Services.
 
 # Prerequisites:
-    Contributor and higher privileges on the App Services in a Subscription.
+    1. Contributor and higher privileges on the App Services in a Subscription.
```
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1:12]: Contributor or higher

File: Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1 [hunk 3]
```diff
@@ -37,30 +38,28 @@
...
         To know more about the options supported by the remediation command, execute:
-        
         Get-Help Enable-HttpsForAppServices -Detailed
 
     To roll back:
         1. To disable HTTPS on the production slot and all non-production slots of all App Services in a Subscription, from a previously taken snapshot:
            Disable-HttpsForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202109131040\EnableHTTPSForAppServices\RemediatedAppServices.csv
         
-        2. To disable HTTPS on the production slot and all non-production slots of all App Services in a Subscription, from a previously taken snapshot:
+        2. To disable HTTPS on the production slot of all App Services in a Subscription, from a previously taken snapshot:
```
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1:59]: why we removed non-production slot comments

File: Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1 [hunk 4]
```diff
@@ -216,18 +221,9 @@ function Enable-HttpsForAppServices
...
-    Write-Host "Checking if $($context.Account.Id) is allowed to run this script..."
-
-    # Checking if the current account type is "User"
-    if ($context.Account.Type -ne "User")
-    {
-        Write-Host "WARNING: This script can only be run by `User` Account Type. Account Type of $($context.Account.Id) is: $($context.Account.Type)" -ForegroundColor $([Constants]::MessageType.Warning)
-        break
-    }
-
     Write-Host "*** To enable HTTPS for App Services in a Subscription, Contributor and higher privileges on the App Services are required. ***" -ForegroundColor $([Constants]::MessageType.Info)
```
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1:224]: In this script we don't have check for permission like minimum Contributor?

File: Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1 [hunk 5]
```diff
@@ -201,9 +207,8 @@ function Enable-HttpsForAppServices
 
     if ([String]::IsNullOrWhiteSpace($context))
     {
-        Write-Host "Connecting to Azure account..."
-        Connect-AzAccount -Subscription $SubscriptionId -ErrorAction Stop | Out-Null
-        Write-Host "Connected to Azure account." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host "AzAccount is not connected. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
```
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1:210]: No active Azure login session found. Exiting...

See if this looks better

File: Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1 (hunk 1 above)
Comment (Aditi-sha) [reply Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1:13]: Updated.

File: Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1 (hunk 2 above)
Comment (Aditi-sha) [reply Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1:12]: Done.

File: Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1 (hunk 4 above)
Comment (Aditi-sha) [reply Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1:224]: No, we do not check for each level (resource/RG/subscription).
We are just giving a warning message.
If user doesn't have permission, Set command will throw an error.

File: Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1 (hunk 3 above)
Comment (Aditi-sha) [reply Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1:59]: In the below command, we are passing a switch -ExcludeNonProductionSlots to exclude non-production slots.

File: Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1 (hunk 5 above)
Comment (Aditi-sha) [reply Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1:210]: Makes sense. 
Updated.

**Review by TarunKrShukla — APPROVED** [review]
Changes look good.

**Review by VishalHaibatpure — APPROVED** [review]

**Pull request merged**